Update main project file to explain the community work and use standardized tools

Trim opened the pull request:
Hello,

As 4.0.0-beta4 seems to be enough stable, I took time to update general files (README, install.rdf, ...) to speak more about our project.

I've decided too to remove home made build and translation tools to use standardized tools.
Currently, there's only Makefile which is new for building project.

I'm looking to use translate-toolkit tools (moz2po, po2moz) to use GNU gettext tools for translations.
With such tools, translators will be able to use their own tool and maybe we will be able to use web services like suggested in #52 

Please, comment this pull request to add your opinion.
Currently, they are just mines.

mtbc commented on CHANGELOG.md:
```diff
@@ -28,7 +38,21 @@ All notable changes to this project will be documented in this file.
+
+### Known bugs
+- Event/Item editor dialog:
+  - Some times, the HTML editor for task content is freezed and nothing can be edited.
```
frozen

mtbc commented on README.md:
```diff
@@ -1,19 +1,131 @@
+  * Lightning extension corresponding to the Thunderbird release
+  * The Exchange server has to provide an Exchange Web Service
+
+This extension was developped by its original author (Michel Verbraak) for
```
developed

mtbc commented on README.md:
```diff
@@ -1,19 +1,131 @@
+This project is a community driven effort to keep maintained and uptodate
+the "Exchange EWS Provider" extension created by Michel Verbraak.
+
+Currently, the community is really small and have too few developpers to
```
developers

mtbc commented on README.md:
```diff
@@ -1,19 +1,131 @@
+
+### Provide feedbak, report issue
+
+You are welcome to provide feedbak on our github project:
```
feedback

mtbc commented on README.md:
```diff
@@ -1,19 +1,131 @@
+* For Javascript files, use tabulations to indent your code
+* For XML files, use spaces
+
+That's a bit strange, we know, but its like the original code has been written.
```
it's

BenjamenMeyer commented on README.md:
```diff
@@ -1,19 +1,131 @@
+Exchange 2007, 2010 and 2013.
+
+Now, ExchangeCalendar community try to keep this extension working, but can't
+provide any guarantee on which versions are supported.
```
grammar, but this would be better as:

```
The ExchangeCalendar community is now maintaining this extension. Support is best effort and cannot be guaranteed. Contributions are welcome.
```

Trim replied:
Thanks, that's better indeed 👍

BenjamenMeyer commented on install.rdf:
```diff
@@ -1,40 +1,27 @@
-    <em:id>exchangecalendar@extensions.1st-setup.nl</em:id>
-    <em:version>4.0.0-beta4</em:version>
+    <em:id>exchangecalendar@ExchangeCalendar</em:id>
+ <em:version>4.0.0-beta5</em:version>
```
note: it would be neat if this could be a template that could be filled in at build time so the version data, etc could be easily kept in sync.

Trim replied:
Now, at build time, the `<em:version>` line is directly filled with the content of the new `VERSION` file (that's work of the `sed` part of the new Makefile). Do you think about something else ?

For the `<em:id>` part, I am not sure that's a good idea to update it.
Indeed, I think that Thunderbird won't erase older addons (the one from 1st-setup and Ericsson) if we modify it.

BenjamenMeyer commented:
@mtbc awesome - thanks for helping standardize this; it'll really help with community growth.

advancingu commented:
@Trim I'm in the process of setting up a free OSS project account with Transifex for translation management as they offer Mozilla DTD support.

Trim commented:
Thanks to have taken contacts with Transifex 👍 

Indeed, they say they have `DTD` support, so we certainly won't need to use the `moz2po` and `po2moz` tools.

Let us know if you need help to adapt the repository to the Transifex prerequisites.

Trim commented on README.md:
```diff
@@ -1,19 +1,131 @@
+That's a bit strange, we know, but it's like the original code has been written.
+
+As current code indentation is really bad and as we are humans, if you edit a
+file, please first use formatters to automatically have better code indentation:
```
```diff
+As current code indentation is really bad and as we are humans, if you edit a
+file, please first use formatters to automatically have better code indentation:
```
I think that's not correctly translate what I mean. What do you think about the following sentence to replace this one?

> The code from the original project is not very well indented and its code style is not consistent. So, when you begin to work on a code that ExchangeCalendar community has not modified recently (you can know this either by looking for last commits date (before August 2017) or by looking the overall indentation and code style), please first use a code beautifier/formatter and commit directly modifications with a message like "auto formatting". Also, with new code, we will be grateful if you can parse your code to such a tool too before sending your patch.

advancingu replied:
Wouldn't it simply be better to fix all the formatting issues in one big PR by simply iterating over all JS files with a script? This will of course break all first level git-blame investigations but the positive aspect is that all future contributors won't have to worry about existing code and PRs won't be a huge mix of formatting fixes + actual semantic changes.
In other words, I think this project will proceed faster, the easier we make it to contribute.

Trim replied:
Yes, indeed, it will help.

The XML formater loose every blank line, but that's then quiet easy to add after files are well structured.

If we do that, we could also:

- take decision between space or tab indentation, so every files have same. I'm in favor of spaces fir this eternal question.
- move files in a clearer directory scheme (I've seen some xul files inside `interface` instead of `chrome`).

advancingu replied:
Regarding indentation, personally I find space indentation with four spaces much easier to maintain than using tab characters.
As for the folder structure, I've replied to you in the other thread.

advancingu commented:
@Trim If I did my research right, you should by now have received an email invitation to Transifex as Project maintainer. Could you try to sync the project up so that translations don't have to be transferred manually? I've found their guide for Github here: https://docs.transifex.com/integrations/github

Trim commented:
Yes, I've well received the invitation. I'm reading the documentation.

Trim commented:
I had time to understand how Transifex work and to update the exchangecalendar project there.

I just wait we merge the modifications made on Github to update the Readme in this pull request to give link to our Transifex project.

advancingu commented:
Code is merged now.

Trim commented:
Ok, I've just rebased my branch with transifex merged code.

Since then, I've added 3 commits which update the README to:
1. Fix typo in Github name
2. Update the patching code to remove advises to use beautifiers and add advice about indentation
3. Update translation section to add Transifex project

You can see these changes herre: https://github.com/ExchangeCalendar/exchangecalendar/pull/70/files/3bfa7acf545f0730b327927b7efa695b2e8ecb8e..31a79bedefee574c4c84d06f41f376b538f7dfaf

Trim commented:
As this PR is still not merged, I've decided to split it into 3 parts to simplify review process.

If you want to add other comments, please continue discussion on #97 #98 and #99
